### PR TITLE
[Static reference] LPS-182671 Remove static references in class MBDiscussionPermission | part 2

### PR DESCRIPTION
**Pull request by W2jx**
Hi @yuhai , I have sent the last pr about MBDiscussionPermission.

### CODE REVIEW

**Comment by W2jx** [comment]
ci:test:lima

**Comment by W2jx** [comment]
ci:test:comment

**Comment by W2jx** [comment]
ci:test:message-boards

**Comment by yuhai** [comment]
@W2jx, I think the current MBDiscussionPermissionImpl is not good.

For the method https://github.com/liferay-core-infra/liferay-portal/blob/fd054898924a6764a0cef88479ef8592fd9ebae7/modules/apps/message-boards/message-boards-comment/src/main/java/com/liferay/message/boards/comment/internal/MBDiscussionPermissionImpl.java#L56-L71.

If checking our portal's Comment interface implementions, they are all MBCommentImpl. That is to say, the method still return 66 line. I think we may change params **Comment for MBMessage, commentId for messageId** in DiscussionPermission interface. MBDiscussionPermissionImpl will look simpler.

**Comment by W2jx** [comment]
Hi @yuhai , I have modified the code.

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:sf

**Comment by W2jx** [comment]
ci:test:lima

**Comment by W2jx** [comment]
ci:test:comment

**Comment by W2jx** [comment]
ci:test:message-boards

**Comment by W2jx** [comment]
ci:stop

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:sf

**Comment by W2jx** [comment]
ci:test:lima

**Comment by W2jx** [comment]
ci:test:comment

File: modules/apps/message-boards/message-boards-comment/src/main/java/com/liferay/message/boards/comment/internal/security/permission/resource/MBDiscussionModelResourcePermission.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	public boolean contains(
+			PermissionChecker permissionChecker, MBDiscussion mbDiscussion,
+			String actionId)
+		throws PortalException {
+
+		return _discussionPermission.hasPermission(
+			permissionChecker, mbDiscussion.getCompanyId(),
+			mbDiscussion.getGroupId(), mbDiscussion.getClassName(),
+			mbDiscussion.getClassPK(), actionId);
+	}
```
Comment (yuhai) [inline modules/apps/message-boards/message-boards-comment/src/main/java/com/liferay/message/boards/comment/internal/security/permission/resource/MBDiscussionModelResourcePermission.java:84]: @W2jx,  `_discussionPermission.hasPermission` doesn't throw exception, please remove `throws PortalException` from contains(PermissionChecker, MBDiscussion, String ). 

In addition, MBDiscussionPermissionImpl#hasPermission(PermissionChecker, Comment, String) and _hasPermission() also needn't to throw Exception.

**Comment by yuhai** [comment]
ci:stop

**Comment by catarineaguiar** [comment]
no regressions found.

SF is not related

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:sf

**Comment by W2jx** [comment]
ci:test:lima

**Comment by W2jx** [comment]
ci:test:comment

**Comment by W2jx** [comment]
ci:test:message-boards

**Comment by W2jx** [comment]
ci:stop

**Pull request closed without merging**